deploy/helm/sumologic/templates/metrics-statefulset.yaml

```diff
@@ -24,6 +24,18 @@ spec:
   replicas: {{ .Values.fluentd.metrics.statefulset.replicaCount }}
   template:
     metadata:
+{{- if or .Values.sumologic.podAnnotations .Values.fluentd.podAnnotations .Values.fluentd.metrics.statefulset.podAnnotations }}
```

Comment:
Can we skip that like for otelcol?
If empty `annotations` is processed correctly I think there is no need for that

Reply:
I've tested it locally with helm and didn't get any errors so I've removed the check. Thanks!